pkg/utils/proto.go

```diff
@@ -0,0 +1,44 @@
+package utils
+
+import (
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// ProtoToYAML converts a Proto message to it's YAML representation in bytes
+func ProtoToYAML(m protoreflect.ProtoMessage) ([]byte, error) {
```

Comment:
Can we add tests for proto.go?

Reply:
It gets implicitly tested in `envoy_config_test.go`, though I agree there should be tests for this file. I consider it outside the scope of this change because this change only moves the function to a different package, which had no test for these functions.

Reply:
Added a test for the function in https://github.com/openservicemesh/osm/pull/3535.